demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+import json
+import logging
+import traceback
+from io import StringIO
+from pathlib import Path
+from typing import List
+
+import pandas as pd
+import typer
+
+from demisto_sdk.commands.common.constants import (
+    FILETYPE_TO_DEFAULT_FROMVERSION,
+    GENERAL_DEFAULT_FROMVERSION,
+    FileType,
+)
+from demisto_sdk.commands.common.logger import setup_rich_logging
+from demisto_sdk.commands.common.tools import get_max_version
+
+app = typer.Typer()
+logger = logging.getLogger("demisto-sdk")
+
+
+@app.command(no_args_is_help=True)
+def generate_modeling_rules(
+    mapping: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=(
+            "The Path to a csv or tsv file containing the mapping for the modeling rules."
+        ),
+    ),
+    raw_event_path: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=("The path to a raw event from the api call in a json format."),
+    ),
+    output_path: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=(
+            "A path to the pack you want to generate the modeling rules in. Best practice to put the working pack path"
+        ),
+    ),
+    vendor: str = typer.Argument(
+        default="test",
+        show_default=False,
+        help=("The vendor name of the product"),
+    ),
+    product: str = typer.Argument(
+        default="test",
+        show_default=False,
+        help=("The vendor name of the product"),
+    ),
+    verbosity: int = typer.Option(
+        0,
+        "-v",
+        "--verbose",
+        clamp=True,
+        max=3,
+        show_default=True,
+        help="Verbosity level -v / -vv / .. / -vvv",
+        rich_help_panel="Logging Configuration",
+    ),
+    quiet: bool = typer.Option(
+        False,
+        help="Quiet output - sets verbosity to default.",
+        rich_help_panel="Logging Configuration",
+    ),
+    log_path: Path = typer.Option(
+        None,
+        "-lp",
+        "--log-path",
+        resolve_path=True,
```

Comment:
Please have @tkatzir review the logs part, to ensure consistency across the SDK logs

Reply:
I will talk to him.